### ISSUE TITLE
Fails to check doctests

### ISSUE THREAD

**Issue by ldicarlo** [haskus/packages#46]
Currently this package [prevent people from running ghcup on nixos](https://github.com/NixOS/nixpkgs/issues/212716).

I think the error message expected in DocTests changed. 

I am trying to prepare a PR.

**Comment by hsyl20** [comment]
It's likely due to https://github.com/haskus/packages/commit/59bf718e5dd1d8ca63046ded07e90e53f659d562

I guess the output now depends on the version of haskus-utils-types used.

**Comment by hsyl20** [comment]
If we can fix it to use the newer message, let's do it. It has been stable for several years already now.

If it's not possible I don't mind if we disable this test and convert it into unchecked documentation instead.

**Comment by ldicarlo** [comment]
I tried to do it but I need some guidance.
I have a lot of doctests failing siren running `stsck test` and using import does not always work. 
Can you tell me how we should do?

**Comment by hsyl20** [comment]
I've tested with the old `stack.yaml` and it passes.

```
stack --stack-yaml stack-8.10.yaml test haskus-utils-variant
```

I've tried with different compilers:

- lts-18.28 (ghc 8.10.7): pass
- lts-19.9 (ghc 9.0.x): fail to find V
- lts-19.33 (ghc-9.0.2): fail to find V
- lts-20.11 (ghc-9.2.5): fail with the error you get

 I've pushed a fix. Strangely enough, it still passes with 8.10 while I expected it to break.
